File: codes/3.ProgrammingLanguage/编程语言试水/Score100WordsAnalyzer.java
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.List;
import java.util.stream.Collectors;

import static java.nio.charset.StandardCharsets.UTF_8;

public class Score100WordsAnalyzer {
  public static void main(String[] args) throws Exception {
    long start = System.currentTimeMillis();
    List<String> score100Words = Files.readAllLines(Paths.get("Dictionary.txt"), UTF_8)
                                      .stream()
                                      .filter(line -> !line.contains("-")
                                          && !line.contains(" ")
                                          && !line.contains("="))
                                      .filter(word -> {
                                        int scoreSum = 0;
                                        for (char c : word.toCharArray()) {
                                          scoreSum += c - 'a' + 1;
                                        }
                                        return scoreSum == 100;
                                      })
                                      .collect(Collectors.toList());
    System.out.println(score100Words);
    System.out.println("\nwords number: " + score100Words.size());
    System.out.println("        time: " + (System.currentTimeMillis() - start) + "ms");
  }
}

File: codes/3.ProgrammingLanguage/编程语言试水/Score100WordsAnalyzer.kt
import java.io.File
import kotlin.streams.toList
import kotlin.system.measureTimeMillis

fun main() {
    val time = measureTimeMillis {
        val score100Words = File("Dictionary.txt")
            .readLines()
            .stream()
            .filter { line ->
                !line.contains("-") && !line.contains(" ") && !line.contains("=")
            }
            .filter { word -> word.sumOf { c -> c - 'a' + 1 } == 100 }
            .toList()
        println(score100Words)
        println("\nwords number: ${score100Words.size}")
    }
    println("        time: $time ms")
}
/**
 * 编码体验：
 * 1、没有分号
 * 2、不需要声明 public class <文件名>
 * 3、main() 方法的声明更简洁了
 * 4、没有 new 关键字
 * 5、扩展了 String 类，sumOf() 方法
 * 6、扩展了 File 类，readLines() 方法
 * 7、不用写 System.out.println()，直接 println()
 * 8、字符串支持模板，不用写 + 号拼接字符串
 */
